Jedwards/asyncio

jedwards4b opened the pull request:
Currently mosart initializes and try's to use IO in the advertise phase, but asyncio requires that this is not done until the realize phase.   This PR splits the mosart initialization so that the namelist is still read in the advertise phase but IO waits until the realize phase.   Tested with cesm prealpha tests on cheyenne, a modification was also required in the mct driver to handle the new initialize subroutine.

ekluzek commented on src/riverroute/RtmMod.F90:
```diff
@@ -317,7 +274,62 @@ subroutine Rtmini(flood_active)
     end do
+  end subroutine Rtminit_namelist
+
+  subroutine Rtmini
```
Add some description to what this is doing, and some of the normal boilerplate comments used by other subroutines to introduce a new subroutine.

ekluzek commented on src/riverroute/RtmMod.F90:
```diff
@@ -130,56 +131,17 @@ subroutine Rtmini(flood_active)
-    character(len=256):: locfn                ! local file name
-    character(len=16384) :: rList             ! list of fields for SM multiply
     integer           :: unitn                ! unit for namelist file
 #ifdef NDEBUG
```
It looks like the NDEBUG setting can be removed from this one. Possibly other things can be removed?

ekluzek reviewed (changes requested):
This is good thanks for working on this and getting a PR for it. The only thing I want you to do is to add some comments that introduce the new subroutine you added similar to the comments for the other subroutines. You can at least get rid of dbug from one of the subroutines, and maybe there are other variables that could also be, that was the one I noticed.

I imagine this will need to happen in RTM and CTSM as well right? And mizuRoute for that matter.

jedwards4b commented:
> I imagine this will need to happen in RTM and CTSM as well right? And mizuRoute for that matter.
No - I only found a couple of components that did the IO initialization this early - mosart and cice.

I've cleaned up and removed unused variables.

jedwards4b commented:
@ekluzek I don't believe that there is a test of the mizuRoute in the cesm test suite.  The key is to make sure no IO is performed until the realize phase of initializaton.

ekluzek commented:
@jedwards4b I took a look at the mizuRoute code, and I didn't see any IO before the realize phase. So it's probably OK, but we might find out differently when we add more testing with mizuRoute in the system.
